tests/CommonTestUtils/GitModuleTestSnapshot.cs

```diff
@@ -0,0 +1,56 @@
+﻿namespace CommonTestUtils;
+
+public class GitModuleTestSnapshot
+{
+    private readonly string _repositoryName;
+    private readonly List<string> _folders = [];
+    private readonly List<(string FilePath, byte[] Content)> _files = [];
+
+    public GitModuleTestSnapshot(string repositoryName, string path)
+    {
+        _repositoryName = repositoryName;
+
+        Populate(new(path), "");
+    }
+
+    public GitModuleTestHelper Clone()
+    {
+        return Clone(FileSystemUtility.GetTemporaryPath());
+    }
+
+    public GitModuleTestHelper Clone(string path)
+    {
+        foreach (string folderPath in _folders)
+        {
+            Directory.CreateDirectory(Path.Combine(path, folderPath));
+        }
+
+        foreach ((string filePath, byte[] content) in _files)
+        {
+            File.WriteAllBytes(
+                Path.Combine(path, filePath),
+                content);
```

Comment:
args are not interesting
```suggestion
            File.WriteAllBytes(Path.Combine(path, filePath), content);
```